app/scripts/app.js

```diff
@@ -28,15 +35,58 @@ blocChat.factory("chatRooms", [
             });
         }
 
+        function setDefaultRoom(name){
+            var roomUrl = 'https://stewart-bloc-chat.firebaseio.com/messages/' + name;
+            var roomRef = new Firebase(roomUrl);
+            var roomRef = $firebaseArray(roomRef);
+            return roomRef;
+        }
+
+        function getRoom(name, roomId) {
+            CurrentRoom.defaults.name = name;
+            var currentRoom = CurrentRoom.defaults.name;
+            if (currentRoom != 'citizens') {
+                var roomUrl = 'https://stewart-bloc-chat.firebaseio.com/messages/' + currentRoom;
+                console.log(roomId);
+                var roomRef = new Firebase(roomUrl);
+                var roomRef = $firebaseArray(roomRef.orderByChild('roomId').equalTo(roomId));
```

Comment:
@philspitler hey Phil, I'm stuck here. If I just return roomRef without .orderByChild('roomId').equalTo(roomId) then I can display all the messages no problem—caveat that messages in /messages/<room.name>/ that don't have the correct roomId also show up. 

This method correctly returns roomId and also name. 

.orderByChild('roomId') also returns and displays messages, but doesn't appear to affect anything. When I add .equalTo(roomId), then the message don't display. And when I log roomRef including the orderByChild and equalTo methods, the object doesn't contain the messages. 

My understanding is that roomUrl—if I get the 'villains' room—will return messages from https://host-path.com/messages/villains/. And when I create roomRef and use the orderByChild method, specifying 'roomId', the query should only return an object ordered by the specified child key, which would then be equalTo 'roomId' when I chain that method. So the returned query object should only contain messages that match that roomId.

This current code state does not return errors in the console.

You can see the current state of the app here: https://mighty-harbor-3058.herokuapp.com/